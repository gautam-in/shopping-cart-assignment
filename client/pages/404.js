import React, {useEffect} from 'react';
import Link from "next/link";
import {useRouter} from "next/router";

const NotFound= () => {
    const router = useRouter();
    useEffect(() => {
          setTimeout(() => router.push('/'), 3000)

    }, [])
    return (
        <div className="not-found">
            <h2>404 Page Doesn't Exist</h2>
            <p>Go back to <Link href="/"><a>home</a></Link></p>

        </div>
    );
};

export default NotFound;