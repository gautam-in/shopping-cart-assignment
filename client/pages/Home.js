import BannerCarusal from '../components/BannerCarusal'
import CategoryList from '../components/CategoryList'
import Head from 'next/head';
export default function Home() {
    return (
        <div>
            <Head>
             <title>Sabka Bazar - Home Page</title>
            </Head>
           <BannerCarusal />
           <CategoryList />
        </div>
    )
}
