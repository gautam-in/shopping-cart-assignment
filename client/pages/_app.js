import Header from "../components/molecules/Header";
import Footer from "../components/molecules/Footer";
import GlobalStyles from "../global/styles/global";
import SectionWrapper from "../components/atoms/SectionWrapper";

function MyApp({ Component, pageProps }) {
  return (
    <>
      <GlobalStyles />
      <Header />
      <main>
        <SectionWrapper>
          <Component {...pageProps} />
        </SectionWrapper>
      </main>
      <Footer />
    </>
  );
}

export default MyApp;
