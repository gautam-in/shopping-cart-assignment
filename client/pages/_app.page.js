import Head from "next/head";
import Header from "../components/molecules/Header";
import Footer from "../components/molecules/Footer";
import GlobalStyles from "../global/styles/global.styles";
import { CartStateProvider } from "../global/utils/useCart";

function MyApp({ Component, pageProps }) {
  return (
    <>
      <Head>
        <title>
          Sabka Bazaar | Online Grocery Store: Buy Online Grocery from India's
          Best Online Supermarket
        </title>
      </Head>
      <GlobalStyles />
      <CartStateProvider>
        <Header />
        <main>
          <Component {...pageProps} />
        </main>
      </CartStateProvider>
      <Footer />
    </>
  );
}

export default MyApp;
