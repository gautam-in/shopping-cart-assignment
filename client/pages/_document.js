import Document, { Html, Head, Main, NextScript } from 'next/document'

class MyDocument extends Document {
    render() {
        return (
            <Html>
                <Head >
                    <link
                        href="https://fonts.googleapis.com/css?family=Dosis:400,600,700"
                        rel="stylesheet"
                    />
                </Head>
                <body>
                    <Main />
                    <NextScript />
                </body>
            </Html>
        )
    }
}

export default MyDocument