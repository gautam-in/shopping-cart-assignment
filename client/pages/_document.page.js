import Document, { Html, Head, Main, NextScript } from "next/document";
import { ServerStyleSheet } from "styled-components";

class MyDocument extends Document {
  static getServerSideProps({ renderPage }) {
    const sheet = new ServerStyleSheet();
    const page = renderPage(
      (App) => (props) => sheet.collectStyles(<App {...props} />)
    );
    const styleTags = sheet.getStyleElement();
    return { ...page, styleTags };
  }

  render() {
    return (
      <Html lang="en">
        <Head>
          <meta
            name="description"
            content="Online Grocery Shopping : Choose from a wide range of grocery, baby care products, personal care products, fresh fruits and vegetables online. Pay Online and Avail exclusive discounts on various products @ India's Best Online Grocery store.
✔ Best Prices and Offers ✔ Cash on Delivery ✔ Easy Returns"
          />
          <link rel="shortcut icon" href="/favicon.ico" />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export default MyDocument;
