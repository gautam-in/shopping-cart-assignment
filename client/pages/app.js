require("./main.css");

// require("./home/home.js");
//require("./cart/cart.js");
// require("./products/products.js");
