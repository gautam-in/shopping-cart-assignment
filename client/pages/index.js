import Home from "../components/Home";

export default function home() {
  return <Home/>;
}
