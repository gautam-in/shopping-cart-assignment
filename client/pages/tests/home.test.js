import React from "react";
import axios from "../../axios.config";
import { render } from "@testing-library/react";
import Home, { getServerSideProps } from "../home.page";
import { carouselData, categoriesList } from "../pages.mock";

jest.mock("../../axios.config", () => ({
  get: jest.fn(),
}));

describe("Home", () => {
  it("should render the component correctly", () => {
    const HomeComponent = render(<Home categoriesList={categoriesList} />);
    expect(HomeComponent).toMatchSnapshot();
  });

  it("getServerSideProps should run correctly", async () => {
    axios.get.mockImplementationOnce(() =>
      Promise.resolve({ data: carouselData })
    );
    axios.get.mockImplementationOnce(() =>
      Promise.resolve({ data: categoriesList })
    );
    const response = await getServerSideProps();
    expect(response).toEqual({
      props: {
        carouselData,
        categoriesList,
      },
    });
  });
});
