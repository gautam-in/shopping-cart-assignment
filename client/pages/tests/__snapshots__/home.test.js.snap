// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render the component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="SectionWrapperstyles__SectionWrapper-sc-1qizg9n-0 gZaGKd"
      >
        <ul
          class="CategoryListstyles__CategoryListWrapper-sc-11ecke9-0 iIwCWX"
        >
          <li
            class="CategoryItemstyles__CategoryItemWrapper-sc-cixyp4-0 jsaYdI"
          >
            <div>
              <div
                style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="display: block; box-sizing: border-box; padding-top: 62.5%;"
                />
                <noscript />
                <img
                  alt="Beverages"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            <div
              class="CategoryItemstyles__CategoryItemDescription-sc-cixyp4-1 cqrdZa"
            >
              <h2>
                Beverages
              </h2>
              <p>
                Our beverage department will ensure your fridge is always fully stocked. Shop for soda, juice, beer and more. 
              </p>
              <button
                class="Buttonstyles__SubmitButton-sc-5pqk3u-0 cIowWi"
                id="5b675e5e5936635728f9fc30"
                name="Explore Beverages"
                title="Explore Beverages"
              >
                Explore 
                Beverages
              </button>
            </div>
          </li>
          <li
            class="CategoryItemstyles__CategoryItemWrapper-sc-cixyp4-0 jsaYdI"
          >
            <div>
              <div
                style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="display: block; box-sizing: border-box; padding-top: 62.5%;"
                />
                <noscript />
                <img
                  alt="Bakery Cakes and Dairy"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            <div
              class="CategoryItemstyles__CategoryItemDescription-sc-cixyp4-1 cqrdZa"
            >
              <h2>
                Bakery Cakes and Dairy
              </h2>
              <p>
                The best cupcakes, cookies, cakes, pies, cheesecakes, fresh bread, biscotti, muffins, bagels, fresh coffee, milk and more.
              </p>
              <button
                class="Buttonstyles__SubmitButton-sc-5pqk3u-0 cIowWi"
                id="5b6899123d1a866534f516de"
                name="Explore Bakery Cakes and Dairy"
                title="Explore Bakery Cakes and Dairy"
              >
                Explore 
                Bakery Cakes and Dairy
              </button>
            </div>
          </li>
          <li
            class="CategoryItemstyles__CategoryItemWrapper-sc-cixyp4-0 jsaYdI"
          >
            <div>
              <div
                style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="display: block; box-sizing: border-box; padding-top: 62.5%;"
                />
                <noscript />
                <img
                  alt="Beauty and Hygiene"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            <div
              class="CategoryItemstyles__CategoryItemDescription-sc-cixyp4-1 cqrdZa"
            >
              <h2>
                Beauty and Hygiene
              </h2>
              <p>
                Buy beauty and personal care products online in India at best prices.
              </p>
              <button
                class="Buttonstyles__SubmitButton-sc-5pqk3u-0 cIowWi"
                id="5b68994e3d1a866534f516df"
                name="Explore Beauty and Hygiene"
                title="Explore Beauty and Hygiene"
              >
                Explore 
                Beauty and Hygiene
              </button>
            </div>
          </li>
          <li
            class="CategoryItemstyles__CategoryItemWrapper-sc-cixyp4-0 jsaYdI"
          >
            <div>
              <div
                style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="display: block; box-sizing: border-box; padding-top: 62.5%;"
                />
                <noscript />
                <img
                  alt="Baby Care"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            <div
              class="CategoryItemstyles__CategoryItemDescription-sc-cixyp4-1 cqrdZa"
            >
              <h2>
                Baby Care
              </h2>
              <p>
                Shop online for Baby Products, Diapers, Skin Care Products,etc.
              </p>
              <button
                class="Buttonstyles__SubmitButton-sc-5pqk3u-0 cIowWi"
                id="5b6899683d1a866534f516e0"
                name="Explore Baby Care"
                title="Explore Baby Care"
              >
                Explore 
                Baby Care
              </button>
            </div>
          </li>
          <li
            class="CategoryItemstyles__CategoryItemWrapper-sc-cixyp4-0 jsaYdI"
          >
            <div>
              <div
                style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="display: block; box-sizing: border-box; padding-top: 62.5%;"
                />
                <noscript />
                <img
                  alt="Fruits & Vegetables"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            <div
              class="CategoryItemstyles__CategoryItemDescription-sc-cixyp4-1 cqrdZa"
            >
              <h2>
                Fruits & Vegetables
              </h2>
              <p>
                A variety of fresh fruits and vegetables.
              </p>
              <button
                class="Buttonstyles__SubmitButton-sc-5pqk3u-0 cIowWi"
                id="5b6899953d1a866534f516e2"
                name="Explore Fruits & Vegetables"
                title="Explore Fruits & Vegetables"
              >
                Explore 
                Fruits & Vegetables
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="SectionWrapperstyles__SectionWrapper-sc-1qizg9n-0 gZaGKd"
    >
      <ul
        class="CategoryListstyles__CategoryListWrapper-sc-11ecke9-0 iIwCWX"
      >
        <li
          class="CategoryItemstyles__CategoryItemWrapper-sc-cixyp4-0 jsaYdI"
        >
          <div>
            <div
              style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="display: block; box-sizing: border-box; padding-top: 62.5%;"
              />
              <noscript />
              <img
                alt="Beverages"
                data-nimg="true"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
          <div
            class="CategoryItemstyles__CategoryItemDescription-sc-cixyp4-1 cqrdZa"
          >
            <h2>
              Beverages
            </h2>
            <p>
              Our beverage department will ensure your fridge is always fully stocked. Shop for soda, juice, beer and more. 
            </p>
            <button
              class="Buttonstyles__SubmitButton-sc-5pqk3u-0 cIowWi"
              id="5b675e5e5936635728f9fc30"
              name="Explore Beverages"
              title="Explore Beverages"
            >
              Explore 
              Beverages
            </button>
          </div>
        </li>
        <li
          class="CategoryItemstyles__CategoryItemWrapper-sc-cixyp4-0 jsaYdI"
        >
          <div>
            <div
              style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="display: block; box-sizing: border-box; padding-top: 62.5%;"
              />
              <noscript />
              <img
                alt="Bakery Cakes and Dairy"
                data-nimg="true"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
          <div
            class="CategoryItemstyles__CategoryItemDescription-sc-cixyp4-1 cqrdZa"
          >
            <h2>
              Bakery Cakes and Dairy
            </h2>
            <p>
              The best cupcakes, cookies, cakes, pies, cheesecakes, fresh bread, biscotti, muffins, bagels, fresh coffee, milk and more.
            </p>
            <button
              class="Buttonstyles__SubmitButton-sc-5pqk3u-0 cIowWi"
              id="5b6899123d1a866534f516de"
              name="Explore Bakery Cakes and Dairy"
              title="Explore Bakery Cakes and Dairy"
            >
              Explore 
              Bakery Cakes and Dairy
            </button>
          </div>
        </li>
        <li
          class="CategoryItemstyles__CategoryItemWrapper-sc-cixyp4-0 jsaYdI"
        >
          <div>
            <div
              style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="display: block; box-sizing: border-box; padding-top: 62.5%;"
              />
              <noscript />
              <img
                alt="Beauty and Hygiene"
                data-nimg="true"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
          <div
            class="CategoryItemstyles__CategoryItemDescription-sc-cixyp4-1 cqrdZa"
          >
            <h2>
              Beauty and Hygiene
            </h2>
            <p>
              Buy beauty and personal care products online in India at best prices.
            </p>
            <button
              class="Buttonstyles__SubmitButton-sc-5pqk3u-0 cIowWi"
              id="5b68994e3d1a866534f516df"
              name="Explore Beauty and Hygiene"
              title="Explore Beauty and Hygiene"
            >
              Explore 
              Beauty and Hygiene
            </button>
          </div>
        </li>
        <li
          class="CategoryItemstyles__CategoryItemWrapper-sc-cixyp4-0 jsaYdI"
        >
          <div>
            <div
              style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="display: block; box-sizing: border-box; padding-top: 62.5%;"
              />
              <noscript />
              <img
                alt="Baby Care"
                data-nimg="true"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
          <div
            class="CategoryItemstyles__CategoryItemDescription-sc-cixyp4-1 cqrdZa"
          >
            <h2>
              Baby Care
            </h2>
            <p>
              Shop online for Baby Products, Diapers, Skin Care Products,etc.
            </p>
            <button
              class="Buttonstyles__SubmitButton-sc-5pqk3u-0 cIowWi"
              id="5b6899683d1a866534f516e0"
              name="Explore Baby Care"
              title="Explore Baby Care"
            >
              Explore 
              Baby Care
            </button>
          </div>
        </li>
        <li
          class="CategoryItemstyles__CategoryItemWrapper-sc-cixyp4-0 jsaYdI"
        >
          <div>
            <div
              style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="display: block; box-sizing: border-box; padding-top: 62.5%;"
              />
              <noscript />
              <img
                alt="Fruits & Vegetables"
                data-nimg="true"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
          <div
            class="CategoryItemstyles__CategoryItemDescription-sc-cixyp4-1 cqrdZa"
          >
            <h2>
              Fruits & Vegetables
            </h2>
            <p>
              A variety of fresh fruits and vegetables.
            </p>
            <button
              class="Buttonstyles__SubmitButton-sc-5pqk3u-0 cIowWi"
              id="5b6899953d1a866534f516e2"
              name="Explore Fruits & Vegetables"
              title="Explore Fruits & Vegetables"
            >
              Explore 
              Fruits & Vegetables
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
